plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/expression/ImplementCountAll.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.pinot.query.expression;
+
+import io.trino.matching.Captures;
+import io.trino.matching.Pattern;
+import io.trino.plugin.pinot.PinotColumnHandle;
+import io.trino.spi.connector.AggregateFunction;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.Verify.verify;
+import static io.trino.plugin.pinot.query.expression.AggregateFunctionPatterns.basicAggregation;
+import static io.trino.plugin.pinot.query.expression.AggregateFunctionPatterns.functionName;
+import static io.trino.plugin.pinot.query.expression.AggregateFunctionPatterns.inputs;
+import static io.trino.plugin.pinot.query.expression.AggregateFunctionPatterns.outputType;
+import static io.trino.spi.type.BigintType.BIGINT;
+
+/**
+ * Implements {@code count(*)}.
+ */
+public class ImplementCountAll
+        implements AggregateFunctionRule
+{
+    // Extracted from io.trino.plugin.jdbc.expression.ImplementCountAll
+    @Override
+    public Pattern<AggregateFunction> getPattern()
+    {
+        return basicAggregation()
+                .with(functionName().equalTo("count"))
+                .with(inputs().equalTo(List.of()))
+                .with(outputType().equalTo(BIGINT));
+    }
+
+    @Override
+    public Optional<PinotColumnHandle> rewrite(AggregateFunction aggregateFunction, Captures captures, RewriteContext context)
+    {
+        verify(aggregateFunction.getOutputType() == BIGINT);
```

Comment:
the intention of https://github.com/trinodb/trino/blob/d649dca65b14c683814b3d1885118ccb93b1e7a3/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/expression/ImplementCountAll.java#L61 is "make sure this code gets updated, should return type of count(*) change to whatever else"

remove `.with(outputType().equalTo(BIGINT));` from the pattern above, to achieve the same (otherwise this line would be redundant)